sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -504,17 +492,43 @@ def _get_snapshots(
                     ),
                     join_type="left",
                 )
-            elif lock_for_update:
+            else:
+                query = query.select(exp.Null().as_("content"))
```

Comment:
what's going on here?

Reply:
Ensures that we always return the same number of rows regardless if we are hydrating or not